MLSSEP (Mario and Luigi Superstar Saga Editor Plus)

bendevnull opened the pull request:
This version of MLSSE is an enhanced version that includes a standard text GUI for users who cannot use the command themselves. MLSSEP requires Python 2.7 and is not compatible with Python 3 (This is an underlying problem in TextUILib2 which you may fix if you feel it necessary to be compatible with Python 3). You may take the credit for MLSSEP if you'd like. Thank you for your time. :)

bendevnull committed:
Rename MLSSE.py to MLSSEP.py

bendevnull committed:
Upload textuilib2.py

bendevnull committed:
Restore MLSSE.py

bendevnull committed:
Write MLSSEP (Unfinished)

bendevnull committed:
Update for MLSSEP

VisualStudio-zz commented:
If they have the ability to install Python then they will most likely know how to use the commands.

bennyman123abc commented:
Yes but, I saw people complaining that they couldn't use the commands or were simply having trouble trying to use the commands. This simplifies that and hopefully expands the audience this tool can reach to.

VisualStudio-zz commented:
If you can come up with a solution and can prove it works well and covers all the command-line arguments then I'll approve this pull request, otherwise I'm going to close it.

bennyman123abc commented:
That is what I am working on now. I will push a commit once it's in a easily accessible state. I will have to add pages for that though but, that will be fine.

hlixed commented on texuilib2.py:
```diff
@@ -0,0 +1,106 @@
+            header.append(h)
+
+def displayHeader(menu):
+    print header[0] + '- ' + menu.getName()
```
This library can be made python3-compatible by adding parentheses around every print statement. This will still work on python 2, since the parentheses will simply be ignored there! That's literally all you need to do to make textlibui2.py work on python3.

bendevnull replied:
I've already tried this. The problem is with the 'return' statements. Instead of returning something like it's supposed to and does in Python 2.7, it returns None in Python 3. If you can fix it, submit a pull request on the repository for the library. Here is a link: https://github.com/bennyman123abc/textuilib-py

hlixed commented on texuilib2.py:
```diff
@@ -0,0 +1,106 @@
+        for o in menu.menuItems:
+            if i == option - 1:
+                return o.function
+            i += 1
```
Since this is an array, you can simply use indexing to extract the element you want from the array instead of using an unnecessarily slow `for` loop. You'll want to make sure the user has selected a valid element, though!

```
option = input('Option: ')
if option-1 < len(menu.menuItems):
    return menu.menuItems[option-1].function
```

bendevnull replied:
Could you make this a pull request along with the other changes needed?

hlixed replied:
I figured out why it's returning nothing: This for-loop structure is hiding a bug. After the line `option = input('Option: ')`, in python3, `option` is a `str` representing the number entered. In python2, however, input() directly evaluates the expression, so "2" becomes `2`. Since `i`, however, is an `int`, and `2 != '2'`, in py3 this works, and py2 it doesn't.

Using array indexing would help to solve this problem, because it would throw an error if the index is a string.

Commit:
Update textuilib2.py to python3, along with minor text fixes

Commit:
Loop upon invalid option, in a way that actually works

bendevnull committed:
Merge pull request #1 from hlixed/master

Update textuilib2.py to python3, along with minor text fixes

bendevnull committed:
Rename textuilib2.py to textuilib3.py

This will be textuilib3 as textuilib2 is still being used in other programs for compatibility reasons.

bendevnull committed:
Update Import Naming

Updated import to use textuilib3 rather than textuilib2

bendevnull committed:
Update README.md

bendevnull committed:
Update README.md

bendevnull committed:
Merge pull request #2 from bennyman123abc/textuilib3-update

Update to textuilib3

hlixed commented:
This PR is now compatible with python 3.
